enterprise/backend/src/metabase_enterprise/transforms_python/python_runner.clj

```diff
@@ -203,17 +202,16 @@
                                    {:error string-if-error}))
                                string-if-error)))))
 
-;; temporary, we should not need to realize data/events files into memory longer term
 (defn read-output-objects
   "Temporary function that strings/jsons stuff in S3 and returns it for compatibility."
   [{:keys [s3-client bucket-name objects]}]
   (let [{:keys [output output-manifest events]} objects
-        output-content          (s3/read-to-string s3-client bucket-name (:path output) nil)
+        output-content          (s3/read-to-stream s3-client bucket-name (:path output) nil)
         output-manifest-content (s3/read-to-string s3-client bucket-name (:path output-manifest) "{}")
-        events-content          (s3/read-to-string s3-client bucket-name (:path events))]
-    {:output          output-content
+        events-content          (s3/read-to-stream s3-client bucket-name (:path events) nil)]
+    {:output-stream   output-content
      :output-manifest (json/decode+kw output-manifest-content)
-     :events          (mapv json/decode+kw (str/split-lines events-content))}))
+     :events          (some->> events-content io/reader line-seq (map json/decode+kw))}))
```

Comment:
For events IIRC it should be truncated / ensured small python side so the eagerly consuming with mapv could be ok. Otherwise I would consider stream lifetime again and ensuring .close on error.